controller/commentsController.js

```diff
@@ -6,6 +6,34 @@ const mongoose = require("mongoose");
 const CustomError = require("../utils/customError");
 const responseHandler = require("../utils/responseHandler");
 
+const validate = (value) => {
+  if (!mongoose.Types.ObjectId.isValid(value)) {
+    new CustomError(422, "invalid ID");
```

Comment:
Please return by calling next and passing the CustomError to the next function. So I'd suggest that you move this logic inside the upvoteComment function.